http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/AwaitCloseChannelPoolMap.java

```diff
@@ -178,7 +178,7 @@ private Bootstrap createBootstrap(URI poolKey) {
                         .channelFactory(sdkEventLoopGroup.channelFactory())
                         .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, configuration.connectTimeoutMillis())
                         // TODO run some performance tests with and without this.
-                        .remoteAddress(new InetSocketAddress(host, port));
+                        .remoteAddress(InetSocketAddress.createUnresolved(host, port));
```

Comment:
Does this break customers using already-resolved IPs, like 127.0.0.1?

Reply:
Verified this doesn't break IP's given as the host:

```java
        SdkAsyncHttpClient netty = NettyNioAsyncHttpClient.builder()
                .build();

        SdkHttpFullRequest sdkRequest = SdkHttpFullRequest.builder()
                .method(SdkHttpMethod.GET)
                .protocol("http")
                .host("52.216.179.173") // s3.amazonaws.com
                .port(80)
                .encodedPath("/")
                .build();

        AsyncExecuteRequest request = AsyncExecuteRequest.builder()
                .requestContentPublisher(new EmptyPublisher())
                .request(sdkRequest)
                .responseHandler(new SdkAsyncHttpResponseHandler() {
                    @Override
                    public void onHeaders(SdkHttpResponse headers) {
                        System.out.println(headers.headers());
                    }

                    @Override
                    public void onStream(Publisher<ByteBuffer> stream) {
                    }

                    @Override
                    public void onError(Throwable error) {
                    }
                })
                .build();

        netty.execute(request).join();
```

Output:
```
{Content-Length=[0], Date=[Thu, 23 Jan 2020 22:44:57 GMT], Location=[https://aws.amazon.com/s3/], Server=[AmazonS3], x-amz-id-2=[vH+t2xruaRBuhxVk//+lRRmf9Nk55Hv8jht7/37/j+3hHyiXBCDngKZaG8bepYJrRjatmthNOLA=], x-amz-request-id=[1ACB290645C7D6E1]}
```